crossopt/assignment2/treenode.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+
+class TreeNode:
+    def __init__(self, val, par=None):
+        self.left = None
+        self.right = None
+        self.val = val
+        self.parent = par
```

Comment:
why storing the parent and not the height?
Do dyou see a difference between the two?

Reply:
Storing the parent rather than the height makes iterating upwards possible, which in turn makes getting all of the ancestors of a node more simple
Also, storing parents instead of heights hypothetically makes linking trees easier